src/Kafka/Internal/Shared.hs

```diff
@@ -102,6 +105,30 @@ readTimestamp msg =
                RdKafkaTimestampLogAppendTime -> LogAppendTime (Millis ts)
                RdKafkaTimestampNotAvailable  -> NoTimestamp
 
+
+readHeaders :: RdKafkaMessageTPtr -> IO (Either RdKafkaRespErrT Headers)
+readHeaders msg = do
+    (err, headersPtr) <- rdKafkaMessageHeaders msg
+    case err of
+        RdKafkaRespErrNoent -> return $ Right mempty
+        RdKafkaRespErrNoError -> fmap headersFromList <$> extractHeaders headersPtr
+        e -> return $ Left e
+    where extractHeaders ptHeaders =
+            alloca $ \nptr ->
+                alloca $ \vptr ->
+                    alloca $ \szptr ->
+                        let go acc idx = rdKafkaHeaderGetAll ptHeaders idx nptr vptr szptr >>= \case
+                                RdKafkaRespErrNoent -> return $ Right acc
+                                RdKafkaRespErrNoError -> do
+                                    cstr <- peek nptr
+                                    wptr <- peek vptr
+                                    csize <- peek szptr
+                                    hn <- BS.packCString cstr
+                                    hv <- word8PtrToBS (fromIntegral csize) wptr
+                                    go ((hn, hv) : acc) (idx + 1)
+                                _ -> error "Unexpected error code"
```

Comment:
The specification says:

```
 * @returns RD_KAFKA_RESP_ERR_NO_ERROR if headers were returned,
 *          RD_KAFKA_RESP_ERR__NOENT if the message has no headers,
 *          or another error code if the headers could not be parsed.
 *
```

Can we return `Left` in case of that `another error code` instead of failing with an `error`?

Reply:
I _think_ that it can be an overlook at `librdkafka` documentation side because handling "if the headers could not be parsed" in another looks strange.
But OK, we can fix it later.